tests/integration/test_remoted/test_agent_communication/test_shared_configuration.py

```diff
@@ -0,0 +1,126 @@
+
+
+# Set pytest marks.
+pytestmark = [pytest.mark.server, pytest.mark.tier(level=1)]
+
+# Cases metadata and its ids.
+cases_path = Path(TEST_CASES_PATH, 'cases_shared_configuration.yaml')
+config_path = Path(CONFIGS_PATH, 'config_shared_configuration.yaml')
+test_configuration, test_metadata, cases_ids = get_test_cases_data(cases_path)
+test_configuration = load_configuration_template(config_path, test_configuration, test_metadata)
+
+daemons_handler_configuration = {'all_daemons': True}
+
+local_internal_options = {REMOTED_DEBUG: '2'}
+
+def check_queue_monitor(agent, pattern):
+    log_queue_monitor = queue_monitor.QueueMonitor(agent.rcv_msg_queue)
+    log_queue_monitor.start(timeout=60, callback=generate_callback(regex=pattern))
+    assert log_queue_monitor.callback_result
+
+# Test function.
+@pytest.mark.parametrize('test_configuration, test_metadata',  zip(test_configuration, test_metadata), ids=cases_ids)
+def test_shared_configuration(test_configuration, test_metadata, configure_local_internal_options, truncate_monitored_files,
+                            set_wazuh_configuration, daemons_handler, simulate_agents):
+
+    '''
+    description: Check if the manager pushes shared configuration to agents as expected.
+                 For this purpose, the test will create an agent for each protocol within the module test cases. Then,
+                 it will try to send the shared configuration to the agent and then, check if the configuration is
+                 completely pushed.
+                 For example, if Wazuh Manager sends new shared files from group shared folder when the merged.mg
+                 checksum is received from an agent is different than the stored one.
+                 As the test has nothing to do with shared configuration files, we removed those rootcheck txt files
+                 from default agent group to reduce the time required by the test to make the checks.
+
+
+    parameters:
+        - test_configuration
+            type: dict
+            brief: Configuration applied to ossec.conf.
+        - test_metadata:
+            type: dict
+            brief: Test case metadata.
+        - truncate_monitored_files:
+            type: fixture
+            brief: Truncate all the log files and json alerts files before and after the test execution.
+        - configure_local_internal_options:
+            type: fixture
+            brief: Configure the Wazuh local internal options using the values from `local_internal_options`.
+        - daemons_handler:
+            type: fixture
+            brief: Restart service once the test finishes stops the daemons.
+        - simulate_agents
+            type: fixture
+            brief: create agents
+        - set_wazuh_configuration:
+            type: fixture
+            brief: Apply changes to the ossec.conf configuration.
+    '''
+
+
+    agent = simulate_agents[0]
+
+    sender, injector = connect(agent = agent, protocol = test_metadata['protocol'])
```

Comment:
Can you do this in a fixture?

Reply:
I spoke with Tomas and an agreement was reached not to do it due to how the tests were designed.